tests/leaderboards/test_tasks_leaderboard.py

```diff
@@ -0,0 +1,471 @@
+
+    Returns:
+        Dict: The updated results dictionary, including:
+            - 'task_hashes': A dictionary with hashes for each task's aggregated samples.
+            - General configuration data from the `evaluation_tracker`.
+    """
+    samples = results.get("samples")
+    evaluation_tracker.general_config_tracker.log_end_time()
+
+    task_hashes = {}
+    for task_name, task_samples in samples.items():
+        sample_hashes = hash_string(
+            "".join(
+                s["doc_hash"] + s["prompt_hash"] + s["target_hash"]
+                for s in task_samples
+            )
+        )
+        task_hashes[task_name] = sample_hashes
+
+    # update initial results dict with the evaluation tracker data
+    results.update({"task_hashes": task_hashes})
+    results.update(asdict(evaluation_tracker.general_config_tracker))
+    return results
+
+
+def compare_results(
+    reference: Dict[str, Any],
+    observed: Dict[str, Any],
+    config_name: str,
+    module_name: str,
+    recursive: bool = False,
+):
+    """
+    Compare values between the reference and observed dictionaries,
+    checking for equality or approximate equality for floats.
+    Compare complex nested structures (e.g., dictionaries or lists) only if `recursive` is set to True.
+    Raise an error if a key from the reference object is not present in the observed object.
+
+    Args:
+        reference (Dict[str, Any]): The reference dictionary to compare with.
+        observed (Dict[str, Any]): The observed dictionary to compare.
+        config_name (str): The name of the config to display in the error message.
+        module_name (str): The associated name of the subresults to display in the error message.
+        recursive (bool): Whether to compare nested structures.
+
+    Example:
+        reference = {
+            "transformers_version": "4.41.1",
+            "system_instruction_sha": None,
+            "versions": {"drop": 3.0}  # Nested structure, compared only if recursive is set to True
+        }
+        observed = {
+            "transformers_version": "4.41.1",
+            "system_instruction_sha": None,
+            "versions": {"drop": 3.0}  # Nested structure, compared only if recursive is set to True
+        }
+        compare_results(reference, observed, "example_config", "example_module")
+    """
+    if not reference:
+        raise ValueError("Reference dictionary is empty.")
+
+    for key, reference_val in reference.items():
+        if isinstance(reference_val, Dict):
+            if recursive:
+                assert key in observed, (
+                    f"Config: '{config_name} - {module_name}' failed. "
+                    f"Key '{key}' is missing in the observed dictionary."
+                )
+                compare_results(
+                    reference_val, observed[key], config_name, key, recursive
+                )
+                continue
+            else:
+                continue
```

Comment:
```suggestion
```

Add an else instead to make it clearer that you've got 3 cases:
- comparing dict
- comparing float
- comparing other values